src/app/shared/services/user.service.spec.ts

```diff
@@ -0,0 +1,223 @@
+import { TestBed, getTestBed, async, inject } from '@angular/core/testing';
+import {
+  HttpModule,
+  Http,
+  Response,
+  ResponseOptions,
+  XHRBackend,
+  BaseRequestOptions,
+  RequestMethod
+} from '@angular/http';
+import { MockBackend, MockConnection } from '@angular/http/testing';
+import { UserService } from './user.service';
+import { ApiService } from './api.service';
+import { User } from '../models/user'
+
+describe('UserService', () => {
+  let mockBackend: MockBackend;
+
+  beforeEach(() => {
+
+    TestBed.configureTestingModule({
+      imports: [HttpModule],
+      providers: [
+        UserService,
+        ApiService,
+        MockBackend,
+        BaseRequestOptions,
+        {
+          provide: Http,
+          deps: [MockBackend, BaseRequestOptions],
+          useFactory:
+          (backend: XHRBackend, defaultOptions: BaseRequestOptions) => {
+            return new Http(backend, defaultOptions);
+          }
+
+        }
+      ]
+    });
+    mockBackend = getTestBed().get(MockBackend);
+
+  });
+
+  it('should get users', (done: DoneFn) => {
+
+    let userService: UserService;
+
+    const mockResponse = {
+      data: [
+        { id: 1, username: "dtrinh100", first_name: "David", last_name: "Trinh", pic_url: "www.example.com/pic", contributions: [], my_playlist: [] },
```

Comment:
This should be broken up into multiple lines for readability. Same applies to the other, similar lines.